File: packages/python/tests/fixtures/fake_validator_repo_both/validator.py
# Both-languages fake validator (Python side)
from galya_reality.base import ContextWindow, Message, ValidationResult, Validator


class FakeBothValidator(Validator):
    name = "fake-both"
    latency_class = "inline"

    async def index(self, message: Message, context: ContextWindow) -> None:
        return None

    def judge(self, message: Message, context: ContextWindow) -> ValidationResult:
        # Deterministic score from content length for parity testing
        score = min(1.0, len(message.content) / 100.0)
        return ValidationResult(score=score, label="ok", rationale="length-based")

File: packages/ts/test/fixtures/fake-validator-repo-both/validator.js
/** Both-languages fake validator (TypeScript ecosystem fixture as plain JS). */

export class FakeBothValidator {
  name = "fake-both";
  latencyClass = "inline";

  async index(_message, _context) {
    return;
  }

  judge(message, _context) {
    const score = Math.min(1.0, message.content.length / 100.0);
    return { score, label: "ok", rationale: "length-based" };
  }
}

export class SyncJudgeValidator {
  name = "sync-judge";
  latencyClass = "inline";
  static indexed = [];

  async index(message, _context) {
    SyncJudgeValidator.indexed.push(message.content);
  }

  judge(_message, _context) {
    return { score: 0.42, label: "sync" };
  }
}

export class AsyncJudgeValidator {
  name = "async-judge";
  latencyClass = "async";

  async index(_message, _context) {
    return;
  }

  async judge(_message, _context) {
    return { score: 0.88, label: "async" };
  }
}
